src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/resource/kubernetes/custom/GenericResourceFactory.kt

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.intellij.kubernetes.model.resource.kubernetes.custom
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.fabric8.kubernetes.api.model.ObjectMeta
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder
+import java.util.stream.Collectors
+
+object GenericResourceFactory {
+
+	const val ITEMS = "items"
+	const val API_VERSION = "apiVersion"
+	const val KIND = "kind"
+	const val METADATA = "metadata"
+	const val SPEC = "spec"
+	const val CREATION_TIMESTAMP = "creationTimestamp"
+	const val GENERATION = "generation"
+	const val NAME = "name"
+	const val NAMESPACE = "namespace"
+	const val RESOURCE_VERSION = "resourceVersion"
+	const val SELF_LINK = "selfLink"
+	const val UID = "uid"
+
+	fun createResources(resourcesList: Map<String, Any?>): List<GenericResource> {
+		val items = resourcesList[ITEMS] as? List<Map<String, Any?>> ?: return emptyList()
+		return createResources(items)
+	}
+
+	private fun createResources(items: List<Map<String, Any?>>): List<GenericResource> {
+		return items.stream()
+				.map { createResource(it) }
+				.collect(Collectors.toList())
+	}
+
+	private fun createResource(item: Map<String, Any?>): GenericResource {
+		return GenericResource(
+				item[API_VERSION] as? String,
+				item[KIND] as? String,
+				createObjectMetadata(item[METADATA] as? Map<String, Any?>),
+				GenericResourceSpec(item[SPEC] as? Map<String, Any?>))
+	}
+
+	private fun createObjectMetadata(metadata: Map<String, Any?>?): ObjectMeta {
+		if (metadata == null) {
+			return ObjectMetaBuilder().build()
+		}
+		return ObjectMetaBuilder()
+				.withCreationTimestamp(metadata[CREATION_TIMESTAMP] as? String?)
+				.withGeneration(metadata[GENERATION] as? Long?)
+				.withName(metadata[NAME] as? String?)
+				.withNamespace(metadata[NAMESPACE] as? String?)
+				.withResourceVersion(metadata[RESOURCE_VERSION] as? String?)
+				.withSelfLink(metadata[SELF_LINK] as? String?)
+				.withUid(metadata[UID] as? String?)
```

Comment:
I think it is enough to just write `metadata[UID] as? String` or `metadata[UID] as String?`. We should stick to using the same rule everywhere. For instance, above, you used `item[SPEC] as? Map<String, Any?>)`

Reply:
IMHO are not the same thing. `metadata[UID] as? String` would return a `null` value if the cast fails (and not throw a class cast exception - see https://kotlinlang.org/docs/reference/typecasts.html#safe-nullable-cast-operator). `metadata[UID] as String?` would throw a class cast exception if the value is not null but of a different type.